sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/PemKeySource.java

```diff
@@ -91,4 +104,46 @@ private RSAPublicKey extractRsaPublicKey(X509Certificate certificate)
     }
     return publicKey;
   }
+
+  private boolean isPublicKey(URI pemKeyLocation) {
+    try (InputStream is = pemKeyLocation.toURL().openStream()) {
+      String keyContent = readContent(is).trim();
+      return keyContent.startsWith("-----BEGIN PUBLIC KEY-----")
+          && keyContent.endsWith("-----END PUBLIC KEY-----");
+    } catch (IOException e) {
+      throw new KeyLoadFailedException("Failed to read key from " + pemKeyLocation, e);
+    }
+  }
+
+  private RSAPublicKey loadPublicKey(URI pemKeyLocation) {
+    LOG.info("Loading public key for token signature verification from PEM {}", pemKeyLocation);
+    try (InputStream is = pemKeyLocation.toURL().openStream()) {
+      String pemPublicKeyContent = readContent(is);
+      String publicKeyPem =
+          pemPublicKeyContent
+              .replace("-----BEGIN PUBLIC KEY-----", "")
+              .replaceAll("(\\r\\n|\\r|\\n)", "")
+              .replace("-----END PUBLIC KEY-----", "");
```

Comment:
Sad that there seems to be no build-in function that supports this format. We probably don't want to add a library just for this use case so I guess it's fine.

Reply:
I'm not aware of any lib we already have as a dependency in this module. My best guess would be to use Bouncy Castle, but that brings in a lot of stuff.